packages/amplify-graphql-relational-transformer/src/graphql-belongs-to-transformer.ts

```diff
@@ -158,6 +159,14 @@ export class BelongsToTransformer extends TransformerPluginBase {
 
 const validate = (config: BelongsToDirectiveConfiguration, ctx: TransformerContextProvider): void => {
   const { field, object } = config;
+  if (!ctx.transformParameters.allowGen1Patterns) {
+    if (field.type.kind === Kind.NON_NULL_TYPE) {
+      throw new InvalidDirectiveError(`@${BelongsToDirective.name} cannot be used on required fields.`);
+    }
+    if (config.fields) {
+      throw new InvalidDirectiveError(`fields argument on @${BelongsToDirective.name} is deprecated. Use references instead.`);
```

Comment:
nit: same suggestion as above about more informative error message. Please apply the same for other Errors added in this PR.